src/modules/processing/services/newgrounds.js

```diff
@@ -0,0 +1,86 @@
+import { genericUserAgent } from "../../config.js";
+import { cleanString } from "../../sub/utils.js";
+
+export default async function(obj) {
+    // handle video downloads
+    if (obj.type == 'portal') {
+        let req = await fetch(`https://www.newgrounds.com/${obj.type}/video/${obj.id}`, {
```

Comment:
if you're already checking if `obj.type` is "portal" then why are you getting `obj.type` immediately after that? couldn't you simplify it to `https://www.newgrounds.com/portal/video/${obj.id}`?